src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetServiceTest.java

```diff
@@ -53,15 +54,76 @@ public void shouldGenerateApplicantCoverSheet() throws Exception {
         assertThat(document().getUrl(), is(caseDocument.getDocumentUrl()));
     }
 
+    @Test
+    public void shouldGenerateApplicantCoverSheetWithEmptySolicitorAddress() throws Exception {
```

Comment:
To help understand quicker what is happening in the background, you could name those new functions following the structure: should...using...when
For example, here it could be:
shouldGenerateApplicationCoverSheetUsingApplicationAddressWhenSolicitorAddressIsEmpty

Reply:
Change made